server/src/main/java/io/crate/expression/scalar/regex/RegexMatcher.java

```diff
@@ -22,37 +22,54 @@
 package io.crate.expression.scalar.regex;
 
 import javax.annotation.Nullable;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
 
 public class RegexMatcher {
```

Comment:
Part of this function is now used by the `regexp_replace` function, the other by the `regexp_matches` functions. Would it make sense to separate this into two classes, or inline the relevant code as inner static class to the relevant functions?

Or another option would be to have two implementations and the implementation would be retrieved via a static constructor that takes the flag as argument, to avoid the `if (global` cases